core/sql/generator/GenPreCode.cpp

```diff
@@ -154,15 +154,19 @@ static NABoolean processConstHBaseKeys(Generator * generator,
   // convert built-in search key to entries with constants, if possible
   if (skey->areAllKeysConstants(TRUE))
     {
-      ValueIdSet exePreds;
       ValueIdSet nonKeyColumnSet;
       idesc->getNonKeyColumnSet(nonKeyColumnSet);
 
-      // seed exePreds with all predicates, key and non-key
-      // they will be reduced by the method below
-      skey->getKeyPredicates(exePreds);
+      // seed keyPreds with only the full key predicate from skey
+      ValueIdSet keyPreds = skey->getFullKeyPredicates();
+
+      // include executorPreds and selection predicates 
+      // but exclude the full key predicates.
+      ValueIdSet exePreds;
+
       exePreds += executorPreds;
       exePreds += relExpr->getSelectionPred();
+      exePreds.subtractSet(keyPreds);
```

Comment:
I am wondering if this is a good thing to do. There are instances where we want a predicate to be both an executor predicate as well as a key predicate. In certain MDAM disjuncts the key predicate guarantees that a minimal set of rows is scanned but an exe pred has to be evaluated on the same pred again to guarantee correct results. I realize that the comment is vague without an example or two. I can  try to find one if necessary. Also since this PreCodeGen a predicate can be physically implemented differently based on what is available. That may also require us to keep things in the exepred?